File: App/App6/servidor/src/api/programacion_routes.py
from controllers.programacion_controller import Programacion_controller

def programacion_register_routes(app):
    """Define las rutas registradas para las programaciones"""
    # Rutas de programaciones
    app.route("/programaciones/bot/<bot_id>", methods=['GET'])(Programacion_controller.get_programaciones_by_bot)
    app.route("/programaciones", methods=['POST'])(Programacion_controller.add_programacion)
    app.route("/programaciones/<programacion_id>", methods=['GET'])(Programacion_controller.get_programacion)
    app.route("/programaciones/<programacion_id>", methods=['PUT'])(Programacion_controller.update_programacion)
    app.route("/programaciones/<programacion_id>", methods=['DELETE'])(Programacion_controller.delete_programacion)

File: App/App6/Cliente/src/api/ProgramacionRoutes.js
import { api_ip } from "../../Config";
const API = api_ip + '/programaciones'; // Actualiza la ruta base para las programaciones

export const deleteProgramacion = async (programacionId) => {
  await fetch(`${API}/${programacionId}`, {
    method: "DELETE",
  });
};

export const getProgramacionesByBot = async (botId) => {
  const res = await fetch(`${API}/bot/${botId}`, {
    method: "GET",
  });

  return await res.json();
};

export const saveProgramacion = async (newProgramacion) => {
  const res = await fetch(`${API}`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(newProgramacion),
  });
  return await res.json();
};

export const getProgramacion = async (programacionId) => {
  const res = await fetch(`${API}/${programacionId}`);
  const programacion = await res.json();
  return programacion;
};

export const updateProgramacion = async (programacionId, updatedProgramacion) => {
  const res = await fetch(`${API}/${programacionId}`, {
    method: "PUT",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(updatedProgramacion),
  });
  return res;
};
